src/synthesizer/line.py

```diff
@@ -139,124 +217,115 @@ class LineRatios:
+    # Balmer decrement, should be [2.79--2.86] (Te, ne, dependent)
+    # for dust free
+    ratios["BalmerDecrement"] = [[Ha], [Hb]]
 
-        a, b = ab
+    # add reference
+    ratios["N2"] = [["N 2 6583.45A"], [Ha]]
+    # add reference
+    ratios["S2"] = S2 = [["S 2 6730.82A", "S 2 6716.44A"], [Ha]]
+    ratios["O1"] = O1 = [["O 1 6300.30A"], [Ha]]
+    ratios["R2"] = [[O2b], [Hb]]
+    ratios["R3"] = R3 = [[O3r], [Hb]]
+    ratios["R23"] = [O3 + O2, [Hb]]
+    ratios["O32"] = [[O3r], [O2b]]
+    ratios["Ne3O2"] = [["NE 3 3868.76A"], [O2b]]
 
-        if fancy:
-            a = map(get_fancy_line_id, a)
-            b = map(get_fancy_line_id, b)
+    # tuple of available ratios
+    available_ratios = tuple(ratios.keys())
 
-        return f"({'+'.join(a)})/({'+'.join(b)})"
+    # dictionary of diagrams
+    diagrams = {}
 
-    def get_ratio_label(self, ratio_id, fancy=False):
-        """
-        Get a line ratio label
+    # add reference
+    diagrams["OHNO"] = [R3, [["NE 3 3868.76A"], O2]]
 
-        Arguments
-        -------
-        ratio_id
-            a ratio_id where the ratio lines are defined in LineRatios
+    # add reference
+    diagrams["BPT-NII"] = [[["N 2 6583.45A"], [Ha]], R3]
 
-        Returns
-        -------
-        str
-            a label
-        """
+    # add reference
+    # diagrams["VO78-SII"] = [[S2, [Ha]], R3]
 
-        ab = self.ratios[ratio_id]
+    # add reference
+    # diagrams["VO78-OI"] = [[O1, [Ha]], R3]
 
-        return f"{ratio_id}={self.get_ratio_label_(ab, fancy = fancy)}"
+    available_diagrams = tuple(diagrams.keys())
 
 
 class LineCollection:
 
     """
     A class holding a collection of emission lines
 
-    Attributes
-    ----------
-    lines : dictionary of Line objects
+    Arguments
+        lines (dictionary of Line objects)
+            A dictionary of line objects.
 
     Methods
-    -------
 
     """
 
     def __init__(self, lines):
+
         self.lines = lines
-        self.line_ids = list(self.lines.keys())
 
-        """
-        these should be filtered to only show ones that are available
-        for the available line_ids
-        """
+        # create an array of line_ids
+        self.line_ids = np.array(list(self.lines.keys()))
 
         # Atrributes to enable looping
         self._current_ind = 0
         self.nlines = len(self.line_ids)
 
-        self.lineratios = LineRatios()
+        # create list of line wavelengths
+        self.wavelengths = (np.array([
+            line.wavelength.to('Angstrom').value for line in
+            self.lines.values()]) * Angstrom)
+
+        # get the arguments that would sort wavelength
+        sorted_arguments = np.argsort(self.wavelengths)
+
+        # sort the line_ids and wavelengths
+        self.line_ids = self.line_ids[sorted_arguments]
+        self.wavelengths = self.wavelengths[sorted_arguments]
+
+        # include line ratio and diagram definitions dataclass
+        self.lineratios = LineRatios
```

Comment:
I could be wrong but I still think a dataclass needs to be called (instantiated)?

Reply:
This is not actually a dataclass, it's essentially a fancy way of holding some dictionaries, so doesn't need to be called. This would also be painful to turn into a dataclass since you wouldn't be able to do:

```
O3b = "O 3 4958.91A"
O3r = "O 3 5006.84A"
O3 = [O3b, O3r]
```

We could think of something better but I have no desire to do that at the moment.

Reply:
Hmm ok, probably not the place for any form of class then. More a module. Since this can't be changed by the user it'll be easy enough.

Reply:
Actually, it can be a dataclass. Have pushed.